internal/evaluation_target/application_snapshot_image/application_snapshot_image.go

```diff
@@ -70,6 +71,11 @@ type ApplicationSnapshotImage struct {
 	signatures   []output.EntitySignature
 	attestations []attestation.Attestation[in_toto.ProvenanceStatementSLSA02]
 	Evaluators   []evaluator.Evaluator
+	Validators   []validator.ImageValidator
+}
+
+func (a *ApplicationSnapshotImage) ImageReference() name.Reference {
```

Comment:
We should drop `ApplicationSnapshotImage`, doesn't really serve any purpose, instead we should have something like `Artifact`, and an `OCIImage` as an implementation or composition of `Artifact`